pkg/iris/mapping.go

```diff
@@ -56,30 +60,28 @@ func (m *mapper) MapDeployments(deployments *appsv1.DeploymentList) ([]Discovery
 	var deploymentDiscoveryItems []DiscoveryItem
 	for _, deployment := range deployments.Items {
 		deployment.ClusterName = m.ClusterName
-		deploymentItem := DiscoveryItem{
-			ID:      deployment.Namespace + ":" + deployment.Name + "-" + deployment.ClusterName,
-			Scope:   "workspace/" + m.WorkspaceId,
-			Type:    "leanix.vsm.item-discovered.runtimeObject",
-			Source:  "kubernetes/" + m.ClusterName,
-			Time:    deployment.CreationTimestamp.String(),
-			Subject: "deployment/" + deployment.Name,
-			Data:    deployment,
-		}
+		id := fmt.Sprintf("%s:%s-%s", deployment.Namespace, deployment.Name, deployment.ClusterName)
+		Type := "leanix.vsm.item-discovered.runtimeObject"
+		Subject := fmt.Sprintf("deployment/%s", deployment.Name)
+
+		// common attributes
+		scope := fmt.Sprintf("workspace/%s", m.WorkspaceId)
+		Source := fmt.Sprintf("kubernetes/%s#%s", m.ClusterName, m.runId)
+		Time := deployment.CreationTimestamp.String()
+
+		deploymentItem := GenerateDiscoveryItem(id, scope, Type, Source, Time, Subject, deployment)
+
 		var DeploymentData = make(map[string]interface{})
 
 		DeploymentData["clusterName"] = m.ClusterName
 		DeploymentData["name"] = deployment.Namespace + ":" + deployment.Name
 		DeploymentData["type"] = "namespaceBased"
-		softwareArtifactItem := DiscoveryItem{
-			ID:      deployment.Namespace + "_" + deployment.Name,
-			Scope:   "workspace/" + m.WorkspaceId,
-			Type:    "leanix.vsm.item-discovered.kubernetesService",
-			Source:  "kubernetes/" + m.ClusterName,
-			Time:    deployment.CreationTimestamp.String(),
-			Subject: "softwareArtifact/" + deployment.Name,
-			Data:    DeploymentData,
-		}
-		deploymentDiscoveryItems = append(deploymentDiscoveryItems, deploymentItem, softwareArtifactItem)
+		id = fmt.Sprintf("%s-%s", deployment.Namespace, deployment.Name)
+		Type = "leanix.vsm.item-discovered.kubernetesService"
+		Subject = fmt.Sprintf("softwareArtifact/%s", deployment.Name)
+		softwareArtifactItem := GenerateDiscoveryItem(id, scope, Type, Source, Time, Subject, DeploymentData)
```

Comment:
Maybe we can encapsulate that in a separate function to make it more readable

Reply:
same as above